File: data/compilers_comparison/src/examples/second_example/src/func52.c
#include "second_example.h" 
array_t* func52(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array15;
   if (pCounter > 0) {
      array15 = vars->data[--pCounter];
      array15->refC++;
      DEBUG_COPY(array15->id);
   } else {
      array15 = (array_t*)malloc(sizeof(array_t));
      array15->size = 170;
      array15->refC = 1;
      array15->id = 15;
      array15->data = (unsigned int*)malloc(array15->size*sizeof(unsigned int));
      memset(array15->data, 0, array15->size*sizeof(unsigned int));
      DEBUG_NEW(array15->id);
   }
   for (int i = 0; i < array15->size; i++) {
      array15->data[i]--;
   }
   for (int i = 0; i < array15->size; i++) {
      if (array15->data[i] == 5) { 
         return array15;
      }
   }
   return array15;
}

File: data/multilang_comparison/program_nim/src/func52.nim
proc func52*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array28: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array28 = vars.data[pCounter]
      inc array28.refC
      debugCopy(array28.id)
   else:
      array28 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array28.size = 367
      array28.refC = 1
      array28.id = 28
      array28.data = cast[ptr UncheckedArray[uint32]](alloc(array28.size * sizeof(uint32)))
      zeroMem(array28.data, array28.size * sizeof(uint32))
      debugNew(array28.id)
   let loopLimit11 = (50) div (5) + 1
   for loop11 in 0 ..< loopLimit11:
      discard
      for i in 0 ..< array28.size:
         if array28.data[i] == 64:
            return array28
   for i in 0 ..< array28.size:
      dec array28.data[i]
   for i in 0 ..< array28.size:
      if array28.data[i] == 87:
         return array28
   return array28

